### PR TITLE
Varming/5.1

### PR DESCRIPTION
**Pull request by varming**
I tried to run scala with a JDK9 early access release 127 and ran into this:

> Exception in thread "main" java.lang.BootstrapMethodError: java.lang.IncompatibleClassChangeError: Inconsistent constant data for scala/collection/generic/Growable.$anonfun$$plus$plus$eq$1(Lscala/collection/generic/Growable;Ljava/lang/Object;)Lscala/collection/generic/Growable; at index 67
        at scala.collection.generic.Growable.$plus$plus$eq$(Growable.scala:50)
        at scala.collection.mutable.ListBuffer.$plus$plus$eq(ListBuffer.scala:182)
        at scala.collection.mutable.ListBuffer.$plus$plus$eq(ListBuffer.scala:44)
        at scala.collection.TraversableLike.to$(TraversableLike.scala:587)
        at scala.collection.TraversableLike.to(TraversableLike.scala:587)
        at scala.collection.TraversableOnce.toList$(TraversableOnce.scala:294)
        at scala.collection.TraversableOnce.toList(TraversableOnce.scala:294)
        at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:41)
        at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:103)
        at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
Caused by: java.lang.IncompatibleClassChangeError: Inconsistent constant data for scala/collection/generic/Growable.$anonfun$$plus$plus$eq$1(Lscala/collection/generic/Growable;Ljava/lang/Object;)Lscala/collection/generic/Growable; at index 67
        ... 10 more

A similar error has been reported to OpenJDK: https://bugs.openjdk.java.net/browse/JDK-8159375. The solution there was to upgrade to ASM 5.1 which handle method references on interfaces correctly (see revision r1791 and r1792 in the ASM svn repo).

### CODE REVIEW

**Comment by retronym** [comment]
@varming FYI, here's the umbrella issue I'm using to scope out JDK9 support: https://github.com/scala/scala-dev/issues/139

**Comment by varming** [comment]
FYI: The change to ASM was discussed here: https://bugs.openjdk.java.net/browse/JDK-8147755

File: README.md [hunk 1]
```diff
@@ -19,10 +19,10 @@ This makes it easy to see how our fork differs from the official release.
 
 ## Current Version
 
-The current sources are based on the following version of ASM ([browse tags here](http://websvn.ow2.org/listing.php?repname=asm&path=%2Ftags%2F&peg=1748)):
+The current sources are based on the following version of ASM ([browse tags here](http://websvn.ow2.org/listing.php?repname=asm&path=%2Ftags%2F&peg=1827)):
```
Comment (lrytz) [inline README.md]: http://websvn.ow2.org/listing.php?repname=asm&path=%2Ftags%2F works and doesn't have a revision in the url

File: README.md [hunk 2]
```diff
@@ -19,10 +19,10 @@ This makes it easy to see how our fork differs from the official release.
...
 
 ```
-Version 5.0.4, SVN r1779, tags/ASM_5_0_4
+Version 5.1, SVN r1827, tags/ASM_5_1
```
Comment (lrytz) [inline README.md]: i think the correct revision here is 1798, see http://websvn.ow2.org/listing.php?repname=asm&path=%2Ftags%2F

**Comment by lrytz** [comment]
LGTM otherwise, thanks a lot!

**Comment by varming** [comment]
Dear @lrytz,

I addressed your comments from  README.md.

**Comment by lrytz** [comment]
Thanks!

**Comment by lrytz** [comment]
pushed a new release (http://repo2.maven.org/maven2/org/scala-lang/modules/scala-asm/5.1.0-scala-1/) and stared using it (https://github.com/scala/scala/pull/5293).

**Pull request merged**